### ISSUE TITLE
Example

### ISSUE THREAD

**Issue by salvino080-coder** [pytorch/xla#9741]
"""
Minimal MNIST example for Colab / TPU VM using torch_xla.
On Colab: select Runtime -> Change runtime type -> TPU and run this script (or convert to notebook).
"""
import torch
import torch.nn as nn
import torch.optim as optim
from torchvision import datasets, transforms
from torch.utils.data import DataLoader

try:
    import torch_xla.core.xla_model as xm
    from torch_xla.distributed.parallel_loader import MpDeviceLoader
except Exception:
    xm = None
    MpDeviceLoader = None

device = xm.xla_device() if xm is not None else torch.device("cpu")

class Net(nn.Module):
    def __init__(self):
        super().__init__()
        self.net = nn.Sequential(
            nn.Flatten(),
            nn.Linear(28*28, 256),
            nn.ReLU(),
            nn.Linear(256, 10)
        )
    def forward(self, x):
        return self.net(x)

def train_one_epoch(model, loader, opt, loss_fn):
    model.train()
    total = 0.0
    count = 0
    for xb, yb in loader:
        xb, yb = xb.to(device), yb.to(device)
        opt.zero_grad()
        out = model(xb)
        loss = loss_fn(out, yb)
        loss.backward()
        if xm is not None:
            xm.optimizer_step(opt)
            xm.mark_step()
        else:
            opt.step()
        total += loss.item()
        count += 1
    return total / max(1, count)

def main():
    transform = transforms.Compose([transforms.ToTensor(), transforms.Normalize((0.5,), (0.5,))])
    train_ds = datasets.MNIST(".", train=True, download=True, transform=transform)
    train_loader = DataLoader(train_ds, batch_size=128, shuffle=True, num_workers=4)
    model = Net().to(device)
    opt = optim.Adam(model.parameters(), lr=1e-3)
    loss_fn = nn.CrossEntropyLoss()

    # If using XLA with multiple processes, wrap with MpDeviceLoader:
    if MpDeviceLoader is not None:
        para_loader = MpDeviceLoader(train_loader, device)
    else:
        para_loader = train_loader

    print("Training on device:", device)
    for epoch in range(1):
        loss = train_one_epoch(model, para_loader, opt, loss_fn)
        print(f"Epoch {epoch}: loss={loss:.4f}")

if __name__ == "__main__":
    main()
